### ISSUE TITLE
Linux: Use desktop portals to get dark theme information

### ISSUE THREAD

**Issue by badcel** [DevToys-app/DevToys#1381]
>> If I might chime in on this issue:
I noticed you changed the code to check the GTK theme name in order to check for dark mode.
This is not an ideal way to check it. In fact, on a modern Gnome Desktop, it doesn't detect dark mode at all:

![Bildschirmfoto vom 2024-08-16 09-11-12](https://github.com/user-attachments/assets/73ddb9a5-cfed-4e15-94c3-53d184a72e8a)

> To give a few details:
While setting dark mode with the GTK theme name was a popular workaround when there was no proper standard, it still has it flaws. For example, `GTK_THEME` is regarded for use in debugging, not to set a dark theme. And some frameworks, like libadwaita, even ignore it entirely. Which is also why Gnome is not setting the GTK theme name when setting dark mode.

> Instead, the proper way would be to check the [Settings Portal](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Settings.html), part of the XDG Desktop Portals. The dark mode preference of Gnome sets the value read by the portal. And the libadwaita code you've used read the dark mode preference from this portal.

> Since GTK does not include support for a dark mode preference, you would need to read the portal directly. You could do this by directly accessing the DBus interface with a library like [Tmds.Dbus](https://github.com/tmds/Tmds.DBus). There's also a C library called [libportal](https://libportal.org/), but I'm not sure what the easier way would be.

> TLDR: Using the GTK theme name to check for dark mode is not ideal (maybe for a fallback), ideally the Settings portal should be used.

_Originally posted by @CodedOre in https://github.com/DevToys-app/DevToys/issues/1378#issuecomment-2293034566_

**Comment by badcel** [comment]
You don't need an additional DBus library as there is already a DBus implementation inside GLib which is part of Gir.Core.

[I wrote a small sample to detect the color-scheme which works for me on fedora 40](https://github.com/gircore/gir.core/blob/main/src/Samples/Gio-2.0/DBus/ReadDesktopAppereanceColorScheme.cs). Any feedback that it works on other current distributions is very welcome.

**Comment by veler** [comment]
Hi @badcel ,

Thank you soooo much for taking the time to explain all this and to provide a sample. You made my life sooo much easier !

I made a pull request that followed your advices: https://github.com/DevToys-app/DevToys/pull/1382

Have a wonderful day !
